nshlib/nsh_netcmds.c

```diff
@@ -776,6 +776,10 @@ int cmd_ifconfig(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
               gip         = addr.s_addr;
             }
         }
+      else
+        {
+          addr.s_addr = 0;
```

Comment:
Changes looks good.
Also during review I see that most probably `gip = 0;` is not needed at line 767 and compare at line 819 should be changed from `if (gip != 0)` to `if (gip != INADDR_ANY)`. Not related to this PR, but still will post a comment here not to forget.